### ISSUE TITLE
Add MSE regression loss to C++ ML framework

### ISSUE THREAD

**Issue by MarcosAsh** [MarcosAsh/3dFluidDynamicsInC#106]
The framework only has softmax cross-entropy loss which is for classification. The Cd surrogate model needs MSE (mean squared error) for regression. Should be straightforward: loss = mean((pred - target)^2), grad = 2*(pred - target)/N.
